### ISSUE TITLE
Reeclaration of class `IntegrationHook`

### ISSUE THREAD

**Issue by live627** [SimpleMachines/SMF#7856]
https://github.com/SimpleMachines/SMF/blob/76dfb27b28bc580faf3078388a9843141150a782/Sources/Autoloader.php#L45-L49

This error goes away once I remove those lines. And my integration still works.

View any page to get the  error.

**Comment by jdarwood007** [comment]
drop the second parameter on class_exists, any changes?

**Comment by Sesquipedalian** [comment]
Does your integration use the integrate_autoload hook?

**Comment by Sesquipedalian** [comment]
> drop the second parameter on class_exists, any changes?

That would most likely make things worse. Unless the second argument is false, class_exists() calls the autoloader. Doing that from within the autoloader would create a recursive loop.

**Comment by jdarwood007** [comment]
During some testing, I found it to come back false with the second param specified.  I'm not certain why, I think it revolves around case sensitivity again.

**Comment by Sesquipedalian** [comment]
The difference you see when setting the second argument to false not due to case sensitivity. Internally, the class_exists() function first checks whether the requested class is already defined and loaded into memory. If it is, then class_exists() returns true. Otherwise, class_exists() moves on to its second step. By default, that second step is to attempt to autoload the requested class, and then return whether the autoloading attempt was successful. But if the second argument is false, no autoloading attempt is made and thus the function simply returns false.

In other words, setting the second argument to false changes the query from "Is this class defined somewhere?" to "Is this class defined _and already loaded_?"

**Comment by Sesquipedalian** [comment]
@live627: Does your integration use the integrate_autoload hook?

**Comment by live627** [comment]
yes

**Comment by Sesquipedalian** [comment]
Excellent. Then, yes, those lines can go. Please feel free to submit a PR.
